io/fasta2/fasta.go

```diff
@@ -0,0 +1,85 @@
+package fasta2
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+)
+
+// Record is a struct representing a single Record element with a Name and its corresponding Sequence.
+type Record struct {
+	Header   string `json:"header"`
+	Sequence string `json:"sequence"`
+}
+
+// buffer is a utility method to serialize the Record in a buffer.
+func (r Record) buffer() bytes.Buffer {
+	var b bytes.Buffer
+	// grow the buffer to allocate just once, the numbers are in order:
+	// the header + > + \n, the sequence + one \n for each 80 char, the last \n
+	b.Grow(len(r.Header) + 2 + len(r.Sequence) + (len(r.Sequence) % 80) + 1)
+	b.WriteByte('>')
+	b.WriteString(r.Header)
+	for i, c := range r.Sequence {
+		// write the fasta sequence 80 characters at a time
+		if i%80 == 0 {
+			b.WriteByte('\n')
+		}
+		b.WriteRune(c)
+	}
+	b.WriteByte('\n')
+
+	return b
+}
+
+// returns the string representation of a Record.
+func (r Record) String() string {
+	b := r.buffer()
+	return b.String()
+}
+
+// returns the representation of a Record as []byte.
+func (r Record) Bytes() []byte {
+	b := r.buffer()
+	return b.Bytes()
+}
+
+// Writes the Record []byte representation to the passed io.Writer.
+func (r Record) Write(w io.Writer) error {
+	recBytes := r.Bytes()
+	_, err := w.Write(recBytes)
+	if err != nil {
+		return fmt.Errorf("error writing record to io.Writer: %w", err)
+	}
+
+	return nil
+}
```

Comment:
Write transforms the bytes into memory (in the bytes buffer) before actually writing them. Why not just directly write to the io Writer?

Reply:
Yeah and also call it `WriteTo` and implement the `io.WriterTo` interface.
I mostly tried to avoid too many allocations since `io.Writer` takes a `[]byte` as arguments so it need to be serialized fist in a `[]byte`, and there was already a way to get the record a `[]byte`.

I was actually just looking at it right now, i might send a fixup later.

Reply:
Also writing to `byte.Buffer` returns almost always `nil` errors so it's just simpler.

Reply:
I sent a small fixup, I think at this point the `Record.Write()` method can be removes, as the user has a way of getting a `string` and `[]byte` representation of the `Record`.
For bulk writing the `Write()` function is now reusing the same buffer (and buffering the I/O is usually a goodthing™).


It would be cool to have a generic `Record.Write()` method that takes either a `bytes.Buffer` or a `bufio.Writer` so we can just decide to pass the buffer or wrap the writer. I unfortunately am traveling tomorrow until next week so no time this weekend 🙁